features/ALB012_Alignment-vertical-segment-radius-of-curvature.feature

```diff
@@ -0,0 +1,20 @@
+@implementer-agreement
+@ALB
+@version1
+@E00020
+Feature: ALB012 - Alignment vertical segment radius of curvature
+  The rule verifies the 'RadiusOfCurvature' design parameter for vertical alignment segments.
+
+  Background:
+    Given A model with Schema "IFC4.3"
+    Given An IfcAlignmentVertical
+    Given A relationship IfcRelNests from IfcAlignmentVertical to IfcAlignmentSegment and following that
+    Given Its attribute DesignParameters
+
+  Scenario: Validating the absence of curvature radius for constant gradient vertical segment
+    Given PredefinedType != 'ARC' or 'PARABOLICARC'
+    Then The value of attribute RadiusOfCurvature must be empty
```

Comment:
The correspondence of `IfcAlignmentVertical <- nests -> Segment -> IfcAlignmentVerticalSegment` is assessed in ALB002. But we don't have rule dependencies. Are we sure that violations of ALB002 don't cause crashes in ALB012 causing the entire validation task to fail?

Reply:
ALB002 is currently disabled with an item in backlog to be re-activated.  I added a link to this comment to the ALB002 reactivation ticket IVS-75 so that I remember to verify independence of ALB002 and ALB012 in UAT.

Reply:
That appears even worse. Is this logic never checked?

https://github.com/buildingSMART/ifc-gherkin-rules/blob/31df6539adce05ed9a5b4966ac3ee804b2895791/features/ALB002_Alignment-layout.feature#L53-L64

I'm just worried about the AttributeErrors when DesignParameters is not of type IfcAlignmentVerticalSegment

Reply:
I agree, for some reason I thought it was IfcAlignmentVerticalSegment that nests IfcAlignmentVertical and that IfcAlignmentSegment was a supertype.

Do we have an implementation "Given attribute is of type {entity_type}?  I looked briefly earlier and didn't find anything.

Regardless of where this lands I will add additional fail files so that we are certain that the type check is handled correctly for both ALB002 and ALB012.

Reply:
Coming back to the original question:

> Are we sure that violations of ALB002 don't cause crashes in ALB012 causing the entire validation task to fail?

Yes, we are sure now because of the additional Given statement in the feature's Background.